src/Components/Components/src/Routing/RouteEntry.cs

```diff
@@ -25,23 +27,53 @@ public RouteEntry(RouteTemplate template, Type handler, string[] unusedRoutePara
 
         internal void Match(RouteContext context)
         {
-            if (Template.Segments.Length != context.Segments.Length)
+            // If there are no optional segments on the route and the length of the route
+            // and the template do not match, then there is no chance of this matching and
+            // we can bail early.
+            var optionalSegments = Template.Segments.Count(template => template.IsOptional);
```

Comment:
If it's easy to do, you could consider statically computing this value stashing it on `RouteTemplate`. It avoids performing a somewhat expensive operation (iterating over a enumerable + linq + a lambda allocation per route) on a per request basis.

Reply:
Sure, can do.